plugin/evm/vm_test.go

```diff
@@ -3709,3 +3710,27 @@ func TestWaitForEvent(t *testing.T) {
 		})
 	}
 }
+
+func TestGenesisGasLimit(t *testing.T) {
+	ctx, db, genesisBytes, _ := setupGenesis(t, upgradetest.Granite)
+	genesis := &core.Genesis{}
+	require.NoError(t, genesis.UnmarshalJSON(genesisBytes))
+	// change the gas limit in the genesis to be different from the fee config
+	genesis.GasLimit = params.GetExtra(genesis.Config).FeeConfig.GasLimit.Uint64() - 1
+	genesisBytes, err := genesis.MarshalJSON()
+	require.NoError(t, err)
+
+	vm := &VM{}
+	err = vm.Initialize(context.Background(), ctx, db, genesisBytes, []byte{}, []byte{}, []*commonEng.Fx{}, &enginetest.Sender{})
+	// This should fail because the gas limit is different from the fee config
+	require.ErrorContains(t, err, "failed to verify genesis")
+
+	// This should succeed because the gas limit is the same as the fee config
+	genesis.GasLimit = params.GetExtra(genesis.Config).FeeConfig.GasLimit.Uint64()
+	genesisBytes, err = genesis.MarshalJSON()
+	require.NoError(t, err)
+	ctx.Metrics = metrics.NewPrefixGatherer()
+
+	require.NoError(t, vm.Initialize(context.Background(), ctx, db, genesisBytes, []byte{}, []byte{}, []*commonEng.Fx{}, &enginetest.Sender{}))
+	vm.Shutdown(context.Background())
```

Comment:
Here we can assert an for errors in `vm.Shutdown` call:
```suggestion
	require.NoError(t, vm.Shutdown(context.Background()))
```